From 2739cef016106feac1acf3498423817c52a956f8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:20 -0400
Subject: [tools] perf/fs: Add procfs support

Message-id: <1408462094-14194-272-git-send-email-jolsa@redhat.com>
Patchwork-id: 88285
O-Subject: [PATCH RHEL7.1 BZ1131394 271/325] perf fs: Add procfs support
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit a9862418547e818aa5842840aecfa81d733b97e9
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Nov 5 15:14:46 2013 +0100

description
===========
Adding procfs support into fs class.

The interface function:
  const char *procfs__mountpoint(void);

provides existing mountpoint path for procfs.

[ Fixup namespace ]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/fs.c b/tools/perf/util/fs.c
index 77bac4e..f5be1f2 100644
--- a/tools/perf/util/fs.c
+++ b/tools/perf/util/fs.c
@@ -9,6 +9,11 @@ static const char * const sysfs__fs_known_mountpoints[] = {
  0,
 };
 
+static const char * const procfs__known_mountpoints[] = {
+ "/proc",
+ 0,
+};
+
 struct fs {
  const char  *name;
  const char * const *mounts;
@@ -18,7 +23,8 @@ struct fs {
 };
 
 enum {
- FS__SYSFS = 0,
+ FS__SYSFS  = 0,
+ FS__PROCFS = 1,
 };
 
 static struct fs fs__entries[] = {
@@ -27,6 +33,11 @@ static struct fs fs__entries[] = {
   .mounts = sysfs__fs_known_mountpoints,
   .magic = SYSFS_MAGIC,
  },
+ [FS__PROCFS] = {
+  .name = "proc",
+  .mounts = procfs__known_mountpoints,
+  .magic = PROC_SUPER_MAGIC,
+ },
 };
 
 static bool fs__read_mounts(struct fs *fs)
@@ -104,4 +115,5 @@ const char *name##__mountpoint(void) \
  return fs__mountpoint(idx); \
 }
 
-FS__MOUNTPOINT(sysfs, FS__SYSFS);
+FS__MOUNTPOINT(sysfs,  FS__SYSFS);
+FS__MOUNTPOINT(procfs, FS__PROCFS);
diff --git a/tools/perf/util/fs.h b/tools/perf/util/fs.h
index a7561c8..5e09ce1 100644
--- a/tools/perf/util/fs.h
+++ b/tools/perf/util/fs.h
@@ -2,5 +2,6 @@
 #define __PERF_FS
 
 const char *sysfs__mountpoint(void);
+const char *procfs__mountpoint(void);
 
 #endif /* __PERF_FS */
diff --git a/tools/perf/util/include/linux/magic.h b/tools/perf/util/include/linux/magic.h
index 58b64ed..07d63cf 100644
--- a/tools/perf/util/include/linux/magic.h
+++ b/tools/perf/util/include/linux/magic.h
@@ -9,4 +9,8 @@
 #define SYSFS_MAGIC            0x62656572
 #endif
 
+#ifndef PROC_SUPER_MAGIC
+#define PROC_SUPER_MAGIC       0x9fa0
+#endif
+
 #endif
-- 
1.7.1